resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/CharsetDeltaJob.java

```diff
@@ -237,7 +238,7 @@ public void setDisabled(boolean disabled) {
 
 	public void shutdown() {
 		try {
-			// try to prevent execution of this job to avoid prevent "already shutdown.":
+			// try to prevent execution of this job to avoid "already shutdown.":
 			cancel();
 			// if job is already running wait for it to finish:
 			join();
```

Comment:
Why not remove `join()`? The job checks for cancellation, so why should we try to block here?

Reply:
The job throws an exception when the workspace is shut down while the job is running

Reply:
And how `join()` should help here? By avoiding to process further with shutdown before the job finishes? Shouldn't we just catch / ignore the error if the case?

Reply:
The join() ensures the job has finished (either canceled or completetd) before workspace is shutdown. Its pretty reproducibly with breakpoints that each of the commands helps to have an errorfree nonblocking shutdown.